// app/handlers/request-handler.js
var CTRL = require('app/handlers/controller');

function extract(res) {
	return {data : res.body, ip : res.ip};
}

function BAD_REQUEST(res) {
    res.status(400).send('Bad Request');
}

function SERVER_ERROR(res) {
    res.status(500).send('Internal Server Error');
}

function OK(res, msg) {
    res.status(200).send(msg);
}

function message_factory(res) {
	this.res = res;
}
message_factory.prototype = {
	send : function(msg) {
        if (typeof msg === 'undefined' || typeof msg === 'string') {
            msg = JSON.stringify({msg : 'OK'});
        } else if (typeof msg === 'object') {
            msg = JSON.stringify(msg);
        }
        OK(this.res, msg);
	},
    error : function(type) {
        if (type === 0) {
            BAD_REQUEST(this.res);
        } else if (type === 1) {
            SERVER_ERROR(this.res);
        }
    }
}

/*function auth(req, res) {
	var raw = extract(req);
	var msg = new message_factory(res);
	CTRL.auth(raw.data.username, raw.data.token, msg);
}

exports.auth = auth;*/
