// app/procedure.js
var DB = require('app/database');

function count_user_on_username_token(u, t, callback) {
    DB.users.count({'username' : u, 'auth.token' : t, 'auth.active' : true}).exec(callback);
}

function logout_user_on_username_token_active(u, t, callback) {
	DB.users.update({
        'username' : u,
        'auth.token' : t,
        'auth.active' : true
    }, {
        $set : {'auth.active' : false}
    }, {
        multi : false
    }).exec(callback);
}

function deactivate_user_log_on_username_token_active(u, t, callback) {
    DB.auth_logs.update({
        'username' : u,
        'auth.token' : t,
        'auth.active' : true
    }, {
        $set : {'auth.active' : false}
    }, {
        multi : false
    }).exec(callback);
}

function deactivate_user_log_on_username_active(u, t, callback) {
    DB.auth_logs.update({
        'username' : u,
        'auth.active' : true
    }, {
        $set : {'auth.active' : false}
    }, {
        multi : false
    }).exec(callback);
}

function find_user_on_username_password(u, p, callback) {
	DB.users.findOne({'username' : u, 'profile.password' : p}).exec(callback);
}

function find_user_on_username(u, callback) {
	DB.users.findOne({'username' : u}).exec(callback);
}

function find_log_on_username_active(u, callback) {
    DB.auth_logs.findOne({'username' : u, 'auth.active' : true}).exec(callback);
}

function all_games(callback) {
    DB.games.find({}).exec(callback);
}

exports.find_user_on_username_token = find_user_on_username_token;
exports.find_user_on_username_password = find_user_on_username_password;
exports.find_user_on_username = find_user_on_username;
